docs/reference/troubleshooting/troubleshooting-unbalanced-cluster.asciidoc

```diff
@@ -0,0 +1,73 @@
+--------------------------------------------------
+// TESTRESPONSE[s/\d+(\.\d+)?[tgmk]?b/\\d+(\\.\\d+)?[tgmk]?b/ s/46/\\d+/]
+// TESTRESPONSE[s/CSUXak2 himrst/.+/ non_json]
+
+This response contains following components that influence the balancing
+
+* `shards` is the current number of shards allocated to the node
+* `shards.undesired` is the number of shards that needs to be moved to other nodes to finish balancing
+* `disk.indices.forecast` expected disk usage according to the projected shard growth
+* `write_load.forecast` project ingest load associated with this node
+
+A cluster is considered balanced when no shard movements are scheduled (all `shards.undesired` are equal to 0).
+At this state it should have spread workload reasonably (eg. some nodes might have fewer shards but more ingest load).
+
+Some operations such as node restarting or decommissioning or changing cluster allocation settings
+are disruptive and might require multiple shards to move or rebalance.
+
+Shard movement order is not deterministic and mostly determined by the source and target node readiness to move a shard.
+While rebalancing is in progress some nodes might appear busier then others.
+
+When shard is allocated to undesired node it utilizes the resources of the current node instead of target.
+This might cause a hotspot (disk or CPU) when multiple shards are residing on the current node that have not been
+moved to their corresponding targets yet.
+
+If cluster is taking a long time to finish rebalancing you might find following log entries:
+[source,text]
+--------------------------------------------------
+[WARN][o.e.c.r.a.a.DesiredBalanceReconciler] [10%] of assigned shards (10/100) are not on their desired nodes, which exceeds the warn threshold of [10%]
+--------------------------------------------------
+This is not concerning as long as the number of such shards is decreasing and this warning appears occasionally,
+for example after rolling restarts or changing allocation settings.
+
+If cluster is experiencing this warning for extended period of time, it is possible that the desired balance diverged
+too far from the current state. In such case is possible to reset the desired balance using the following API call
+and calculate a new one that would hopefully require less shard movements to get to balanced state.
+
+[source,console,id=delete-desired-balance-request-example]
+--------------------------------------------------
+DELETE /_internal/desired_balance
+--------------------------------------------------
+
+If cluster is getting into this state repeatedly without node restarts nor changes in allocation settings,
+it is possible to increase the <<shards-rebalancing-heuristics,cluster.routing.allocation.balance.threshold>> in order
+to reduce the sensitivity of the algorithm that tries to level up the shard count and disk usage within the cluster.
```

Comment:
```suggestion
If the cluster gets into this state repeatedly, without node restarts or changes in allocation settings,
you can increase the <<shards-rebalancing-heuristics,`cluster.routing.allocation.balance.threshold`>>.
This will reduce the sensitivity of the algorithm that tries to level up the shard count and disk usage within the cluster.
```